docs/design/plan-modification.md

```diff
@@ -0,0 +1,298 @@
+
+### `schema.Attribute.RequiresReplace`
+
+Like `helper/schema`, we could add a `ForceNew bool`, here called `RequiresReplace` to match the protocol, to the framework's `schema.Attribute` struct, enabling provider developers to take advantage of this simple schema behaviour with one line of code.
+
+Precedent for expressing RequiresReplace in a declarative manner is found in the AWS API, which has a concept of _create-only properties_: "properties that are only able to be specified by the customer when creating a resource" (see [CloudFormation Resource Semantics](https://github.com/aws-cloudformation/cloudformation-resource-schema#resource-semantics). For this API at least, marking such properties with ForceNew aligns with the [provider design principle](https://www.terraform.io/docs/extend/hashicorp-provider-design-principles.html#resource-and-attribute-schema-should-closely-match-the-underlying-api): _Resource and attribute schema should closely match the underlying API_.
+
+#### Tradeoffs
+
+A single `RequiresReplace` schema attribute is easily discoverable, Go-native, minimally verbose, and a reasonably transparent representation of the protocol's concept of `RequiresReplace`.
+
+A cursory survey of existing provider code finds `ForceNew` very widely used, and seldom the subject of individual acceptance tests, likely because provider developers do not feel the need to test a behaviour whose implementation resides in the SDK itself. For this reason the question of unit testability is also moot. 
+
+One possible disadvantage of this approach is that it is atomic with respect to compatibility - unlike provider-defined plan modification functions like those described below, the framework must deprecate the field or undergo a breaking change in order to modify the behaviour of `schema.Attribute.RequiresReplace`.
+
+### `schema.Attribute.RequiresReplaceIf`
+
+An field on the `schema.Attribute` struct could add an optional `RequiresReplaceIf` function to schema attributes:
+
+```go
+type Attribute struct {
+  // ...
+  
+  RequiresReplaceIf RequiresReplaceIfFunc 
+}
+
+type RequiresReplaceIfFunc func(context.Context, old, new attr.Value) bool
+```
+
+Provider code:
+
+```go
+type fileResourceType struct{}
+
+// GetSchema returns the schema for this resource.
+func (f fileResourceType) GetSchema(_ context.Context) (schema.Schema, []*tfprotov6.Diagnostic) {
+	return schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"favorite_number": {
+				Type:     types.NumberType,
+				Required: true,
+				RequiresReplaceIf: func(ctx context.Context, old, new attr.Value) bool {
```

Comment:
nitpick: I'd very much like to stay away from "old" or "new" being parameter names. First, `new` is a keyword in Go, and that feels like asking for trouble. Second, `old` and `new` _what_? `state` and `config` may be better names.